crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -133,8 +146,14 @@ impl MsgArea {
             Some(height) => height,
             None => {
                 let mut total_height = 0;
-                for line in &mut self.lines {
-                    total_height += line.rendered_height(self.width);
+                let recalculate_scroll = self.recalculate_scroll();
+                for (idx, line) in &mut self.lines.iter_mut().enumerate() {
+                    let height = line.rendered_height(self.width);
+                    // If we need to recalculate, keep adding line heights until we get to line_idx
+                    if recalculate_scroll && idx >= self.scroll.line_idx {
+                        self.scroll.scroll += height;
+                    }
+                    total_height += height;
```

Comment:
It's a bit surprising that a function that is supposed to return total height of a widget re-calculates scroll value. Why is this done here?

I'm also a bit confused about what this is supposed to return. It's not documented, but it's returning `self.lines_height` so I checked the documentation of it, but that also doesn't make sense. It says `total rendered height of all lines` but (1) why would anyone need to know about the height of all lines, only the visible part should be enough (needs documentation) (2) the code uses the scroll value, which isn't relevant if we're calculating the total height of all lines, because that doesn't change depending on the scroll value.

Reply:
I have renamed the function and added a doc line to make the purpose more clear.
 > (1) why would anyone need to know about the height of all lines, only the visible part should be enough (needs documentation) 

This is for scrolling, specifically to know what number to set the scroll to

> (2) the code uses the scroll value, which isn't relevant if we're calculating the total height of all lines, because that doesn't change depending on the scroll value.

This is for recalculating the scroll so that we know how to adjust the scroll number during resizing (width changes).